### PR TITLE
Deprecate IconButton and replace its usage with Button

### PR DESCRIPTION
**Pull request by youknowriad**
This PR replaces all usage of IconButton by Button in the project and deprecates the IconButton component.

**testing instructions**

 - Tests should pass.

### CODE REVIEW

**Comment by youknowriad** [comment]
Is this safe for mobile @hypest @Tug ? the Button component must support the icon prop here.

File: packages/components/src/icon-button/index.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /**
  * WordPress dependencies
  */
+import deprecated from '@wordpress/deprecated';
```
Comment (gziolo) [inline packages/components/src/icon-button/index.js]: I’m on mobile so I can’t do a proper check. Is README file removed? It should as our docs generator will still keep it exposed. Maybe we could remove the folder completely and put the file in deprecated subfolder?

**Comment by gziolo** [comment]
Nice one, I wasn’t sure in which direction you would go but seeing this PR, I admit I like it. We should follow up and make similar refactoring for the Toolbar components family 👍

File: packages/components/src/icon-button/index.js (hunk 1 above)
Comment (youknowriad) [reply packages/components/src/icon-button/index.js]: Good call

**Comment by gziolo** [comment]
Great one, thanks for wrangling it 💯

**Comment by Soean** [comment]
The IconButton is still in the handbook:
https://developer.wordpress.org/block-editor/components/icon-button/

We need to add the icon prop to the button docs:
https://developer.wordpress.org/block-editor/components/button/

**Pull request merged**